src/strategy/StrategyV1.sol

```diff
@@ -0,0 +1,678 @@
+        governance = owner;
+
+        minter = _minter;
+        veToken = _veToken;
+        locker = ILocker(_locker);
+        rewardToken = _rewardToken;
+    }
+
+    /// @notice Initialize the strategy.
+    /// @param owner Address of the owner.
+    /// @dev The implementation should not be initalized since the constructor already define the governance.
+    function initialize(address owner) external virtual {
+        if (governance != address(0)) revert GOVERNANCE();
+        governance = owner;
+
+        /// Initialize the SDT Distributor.
+        SDTDistributor = 0x9C99dffC1De1AfF7E7C1F36fCdD49063A281e18C;
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- MODIFIERS
+    //////////////////////////////////////////////////////
+
+    modifier onlyVault() {
+        if (!vaults[msg.sender]) revert UNAUTHORIZED();
+        _;
+    }
+
+    modifier onlyGovernance() {
+        if (msg.sender != governance) revert GOVERNANCE();
+        _;
+    }
+
+    modifier onlyGovernanceOrAllowed() {
+        if (msg.sender != governance && !allowed[msg.sender]) revert UNAUTHORIZED();
+        _;
+    }
+
+    modifier onlyGovernanceOrFactory() {
+        if (msg.sender != governance && msg.sender != factory) revert GOVERNANCE_OR_FACTORY();
+        _;
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- DEPOSIT / WITHDRAWAL LOGIC
+    //////////////////////////////////////////////////////
+
+    /// @notice Deposit LP token.
+    /// @param asset Address of LP token to deposit.
+    /// @param amount Amount of LP token to deposit.
+    /// @dev Only callable by approved vaults.
+    function deposit(address asset, uint256 amount) external onlyVault {
+        // Transfer the token to this contract.
+        SafeTransferLib.safeTransferFrom(asset, msg.sender, address(this), amount);
+
+        /// Deposit the token in the locker.
+        _deposit(asset, amount);
+    }
+
+    /// @notice Withdraw LP token.
+    /// @param asset Address of LP token to withdraw.
+    /// @param amount Amount of LP token to withdraw.
+    /// @dev Only callable by approved vaults.
+    function withdraw(address asset, uint256 amount) external onlyVault {
+        /// Withdraw from the locker.
+        _withdraw(asset, amount);
+
+        /// Transfer the token to the vault.
+        SafeTransferLib.safeTransfer(asset, msg.sender, amount);
+    }
+
+    //////////////////////////////////////////////////////
+    /// --- LOCKER FUNCTIONS
+    //////////////////////////////////////////////////////
+
+    /// @notice Deposit into the gauge trhoug the Locker.
+    /// @param asset Address of LP token to deposit.
+    /// @param gauge Address of Liqudity gauge corresponding to LP token.
+    /// @param amount Amount of LP token to deposit.
+    function _depositIntoLocker(address asset, address gauge, uint256 amount) internal virtual {
```

Comment:
I dunno if I prefer `_depositFromLocker` or `_depositUsingLocker` here but `_depositIntoLocker` sounds weird to me considering what the function does (using the locker as the intermediary contract that will call the gauge)